### PR TITLE
Implement `TextChannel.follow()`

### PR DESCRIPTION
**Pull request by NCPlayz**
### Summary

This allows the client to follow a news (announcement) channel.

```py
>>> await my_news_channel.follow(destination=channel)
<Webhook id=:id:> # no token provided by discord
```

### Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by CapnS** [comment]
I feel like this should be `await channel.follow(webhook_channel)` like how other methods are in d.py.

**Comment by NCPlayz** [comment]
Ok! Reversed it so that the destination is provided for the method.

File: discord/webhook.py [hunk 1]
```diff
@@ -495,6 +495,25 @@ def from_url(cls, url, *, adapter):
             raise InvalidArgument('Invalid webhook URL given.')
         return cls(m.groupdict(), adapter=adapter)
 
+    @classmethod
+    def as_follower(cls, data, *, channel, user):
```
Comment (Rapptz) [inline discord/webhook.py]:
```suggestion
    def _as_follower(cls, data, *, channel, user):
```

File: discord/channel.py [hunk 2]
```diff
@@ -455,6 +455,43 @@ def is_me(m):
...
+        --------
+        :class:`Webhook`
+            The created webhook.
+        """
+
+        if not self.is_news():
+            raise ClientException('The channel must be a news channel.')
+
+        data = await self._state.http.follow_webhook(self.id, webhook_channel_id=destination.id)
+        return Webhook.as_follower(data, channel=destination, user=self._state.user)
```
Comment (Rapptz) [inline discord/channel.py]:
```suggestion
        return Webhook._as_follower(data, channel=destination, user=self._state.user)
```

**Pull request closed without merging**